bigchaindb_driver/transport.py

```diff
@@ -21,8 +23,30 @@ def __init__(self, *nodes, headers=None):
 
     def init_pool(self, nodes, headers=None):
         """Initializes the pool of connections."""
+        if isinstance(nodes[0], dict):
+            self.init_nodes_dict(nodes)
```

Comment:
If `nodes` are passed as a dictionary, the `headers` argument is ignored, what might be confusing..

It might be hard to maintain multiple ways of describing nodes so I tried to come up with an alternative that uses a single input format, yet preserving backwards-compatibility. What do you think about it? 

```
def init_pool(self, nodes, headers=None, node_specific_headers=None):
    """Initializes the pool of connections."""
    
    self.pool = Pool([Connection(node_url=node, headers=self._get_node_headers(node))
                      for node in nodes])

def _get_node_headers(self, node, common_headers=None, node_specific_headers=None):
    if common_headers is None:
        common_headers = {}
    if node_specific_headers is None:
        node_specific_headers = {}
    return {**common_headers, **node_specific_headers.get(node, {})}
```

Reply:
to be honest i don't understand what you propose as improvement, and well the signature kind of disagrees with the [BEP14](https://github.com/bigchaindb/BEPs/tree/master/14) signature proposal, because of the new parameter your introduced for adding specific node headers.

on the other hand, i disagree about the current approach to be confusing. Any ways, care to elaborate more?

Reply:
I find it harder to figure out how the function works when 2 different input formats are involved. For example, by looking at the function signature, a user may infer that the function can accept some nodes as plain strings and others as dictionaries. May be he has 10 nodes and wants to specify a custom header for only one of them so he may try to do `init_pool(nodeurl1, nodeurl2, ... nodeurl9, {'node': nodeurl10, 'headers': ..})` what will fail in some weird way w.r.t current implementation. Another point here is if I only want custom headers for a small subset of nodes it would be easier for me to only alter the way I pass this node as a parameter, not all nodes.

What do you think about the following alternative, still using 2 formats, but allowing to only specify a dict when it's really needed?
```
def init_pool(nodes, headers=None):
    connections = []
    for node in nodes:
        if isinstance(node, str):
           connections.append({'node': Connection(node_url=node, headers=headers), 'time': time})
        else:
           node_headers = self._merge_headers(node['headers'], headers)
           connections.append({'node': Connection(node_url=node['endpoint'], headers=node_headers), 'time': time})
```

Reply:
i agree so ill update accordingly on next push.